src/main/java/florizz/command/RecommendCommand.java

```diff
@@ -86,31 +116,40 @@ private Flower.Occasion askOccasion(Ui ui) throws FlorizzException {
      * @param eligibleFlowers list of flowers to choose from
      * @return Colour enum
      */
-    private Flower.Colour askColour(Ui ui, ArrayList<Flower> eligibleFlowers) throws FlorizzException {
+    private Flower.Colour askColour(Ui ui, ArrayList<Flower> eligibleFlowers) {
         assert !eligibleFlowers.isEmpty() : "Eligible flowers should not be empty";
         logger.entering(RecommendCommand.class.getName(), "askColour");
-        String colourInput = Parser.parseColour(ui.printAskColour(eligibleFlowers));
-
-        // check if colour is in our dictionary
-        if (!Flower.isValidColour(colourInput)) {
-            throw new FlorizzException("This colour does not exist. Type 'colour' to get a list of colours");
+        String colourInput = "placeHolder";
+        boolean isValidFormat = false;
+        boolean isValidColour = false;
+        while (!(isValidColour && isValidFormat)) {
+            colourInput = ui.printAskColour(eligibleFlowers);
+            isValidFormat = Parser.parseColour(colourInput);
+            isValidColour = Flower.isValidColour(colourInput);
+
+            // check if colour is in our dictionary
+            if (!isValidColour) {
+                System.out.println("This colour does not exist.");
+            }
         }
 
         logger.exiting(RecommendCommand.class.getName(), "askColour");
         return Flower.stringToColour(colourInput);
     }
 
     private void askSaveBouquet(Ui ui, ArrayList<Bouquet> bouquetList,
-                                Bouquet recommendedBouquet) throws FlorizzException {
+                                Bouquet recommendedBouquet) {
         logger.entering(RecommendCommand.class.getName(), "askSaveBouquet");
-        String saveInput = Parser.parseSaveBouquet(ui.printAskSaveBouquet(recommendedBouquet));
-
-        if (saveInput.equals("yes")) {
-            if (bouquetList.contains(recommendedBouquet)) {
-                // change name of bouquet
-                recommendedBouquet.setName(recommendedBouquet.getBouquetName() + " (1)");
-            }
+        String saveInput = "placeHolder";
+        boolean isValidFormat = false;
+        boolean isValidInput = false;
+        while (!(isValidInput && isValidFormat)) {
+            saveInput = ui.printAskSaveBouquet(recommendedBouquet);
+            isValidFormat = Parser.parseSaveBouquet(saveInput);
+            isValidInput = (saveInput.equalsIgnoreCase("yes") || saveInput.equalsIgnoreCase("no"));
+        }
```

Comment:
`isValidInput` might not be necessary since `parseSaveBouquet` already used regex, although that one only accepts lower case input. If you didn't change `parseSaveBouquet`, the method might still reject capitalised yes even with `equalsIgnoreCase`.